### PR TITLE
Adding iterator

### PR DESCRIPTION
**Pull request by ruvdrsti**
@xwieme 
For some reason I can't push into develop_code directly

### CODE REVIEW

**Comment by ruvdrsti** [comment]
or doing the PR directly into main will work as well I guess

File: presentation.tex [hunk 1]
```diff
@@ -3,7 +3,57 @@
...
 \usepackage[utf8]{inputenc}
 \usepackage{braket}
+\usepackage{listings}
 \usetheme{Copenhagen}
+
+% listings python setting
+
+% Default fixed font does not support bold face
+\DeclareFixedFont{\ttb}{T1}{txtt}{bx}{n}{12} % for bold
+\DeclareFixedFont{\ttm}{T1}{txtt}{m}{n}{12}  % for normal
```
Comment (xwieme) [inline presentation.tex]:
```suggestion
\DeclareFixedFont{\ttb}{T1}{txtt}{bx}{n}{9} % for bold
\DeclareFixedFont{\ttm}{T1}{txtt}{m}{n}{9}  % for normal
```
I find the code cleaner when there are fewer line wraps. What do you think?

File: presentation.tex [hunk 2]
```diff
@@ -149,9 +199,188 @@ \section{Why?}
...
+        """
+        Input:
+            molecule: psi4.core.Molecule
+            
+        Performs the restricted hartree fock iterations and print the progress on screen
+        """
+        
+        iteration = 0
+        iterations = [0]
+        energies = list()
```
Comment (xwieme) [inline presentation.tex]:
```suggestion
def SCF(self):
    iteration = 0
    iterations = [0]
    energies = list()
    # Init
    self.one_electron_integrals()
    self.ElectronRepulsion()
    self.density_matrix(initial=True)
    self.fock_matrix()
    self.electronic_energy()
    self.total_energy()
    prev_D = self.D
    prev_E_elec = self.E_elec
    energies.append(prev_E_elec)
```

File: presentation.tex [hunk 3]
```diff
@@ -149,9 +199,188 @@ \section{Why?}
...
+            delta_D = np.sqrt(np.einsum('ij->', np.square(self.D - prev_D)))
+        \end{python}
+    \end{frame}
+    \begin{frame}[fragile]
+        \begin{python}
+            # Print progress
+            iteration += 1
+            iterations.append(iteration)
+            energies.append(self.E_elec)
+            print("{} {:>15.10f} {:>15.10f} {:>15.10f}".format(iteration, E_elec , delta_E_elec, delta_D))
```
Comment (xwieme) [inline presentation.tex]:
```suggestion
            print("{} {:>15.10f} {:>15.10f} {:>15.10f}".format(iteration, self.E_elec , delta_E_elec, delta_D))

```

File: presentation.tex (hunk 1 above)
Comment (ruvdrsti) [reply presentation.tex]: it does indeed look a lot better, I will commit it

**Pull request merged**